flink-end-to-end-tests/flink-connector-hive-test/src/test/java/org/apache/flink/batch/connectors/hive/LocalExecutorHiveConnectorTest.java

```diff
@@ -0,0 +1,150 @@
+ */
+
+package org.apache.flink.batch.connectors.hive;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.client.cli.util.DummyCustomCommandLine;
+import org.apache.flink.client.program.ClusterClient;
+import org.apache.flink.configuration.ConfigConstants;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.TaskManagerOptions;
+import org.apache.flink.configuration.WebOptions;
+import org.apache.flink.runtime.jobgraph.JobStatus;
+import org.apache.flink.runtime.testutils.MiniClusterResourceConfiguration;
+import org.apache.flink.table.catalog.hive.client.HiveMetastoreClientFactory;
+import org.apache.flink.table.catalog.hive.client.HiveMetastoreClientWrapper;
+import org.apache.flink.table.client.config.Environment;
+import org.apache.flink.table.client.gateway.Executor;
+import org.apache.flink.table.client.gateway.ProgramTargetDescriptor;
+import org.apache.flink.table.client.gateway.SessionContext;
+import org.apache.flink.table.client.gateway.local.LocalExecutor;
+import org.apache.flink.table.client.gateway.utils.EnvironmentFileUtil;
+import org.apache.flink.test.util.MiniClusterWithClientResource;
+import org.apache.flink.util.FileUtils;
+
+import com.klarna.hiverunner.HiveShell;
+import com.klarna.hiverunner.annotations.HiveSQL;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests Hive connector with LocalExecutor.
+ */
+@RunWith(FlinkStandaloneHiveRunner.class)
+public class LocalExecutorHiveConnectorTest {
+
+	private static final int NUM_TMS = 2;
+	private static final int NUM_SLOTS_PER_TM = 2;
+
+	private static Executor executor;
+	private static SessionContext session;
+	private static HiveMetastoreClientWrapper hmsClient;
+
+	private static final String HIVE_CATALOG_YAML_FILE = "test-sql-client-hive.yaml";
+
+	@HiveSQL(files = {})
+	private static HiveShell hiveShell;
+
+	@ClassRule
+	public static final MiniClusterWithClientResource MINI_CLUSTER_RESOURCE = new MiniClusterWithClientResource(
+			new MiniClusterResourceConfiguration.Builder()
+					.setConfiguration(getConfig())
+					.setNumberTaskManagers(NUM_TMS)
+					.setNumberSlotsPerTaskManager(NUM_SLOTS_PER_TM)
+					.build());
+
+	@BeforeClass
+	public static void setup() throws Exception {
+		executor = createHiveExecutor(MINI_CLUSTER_RESOURCE.getClusterClient());
+		session = new SessionContext("test-session", new Environment());
+
+		hmsClient = HiveMetastoreClientFactory.create(hiveShell.getHiveConf(), null);
+	}
+
+	@Test
+	public void testWriteHiveTable() throws Exception {
+
+		// primitive types
+		hiveShell.execute("create database db1");
```

Comment:
End-to-end test should involve more features. I would suggest the following tests:

- List all tables and read some metadata
- read from an existing table
- register a new table in Hive
- use a registered function
- write to an existing table
- write to a new table
- Use all types that Hive provides in tables.

Reply:
True, yes then let's have a e2e test that submits a regular jar packaged program via the Flink command line.